crates/curp/src/server/storage/wal/remover.rs

```diff
@@ -0,0 +1,228 @@
+use std::{
+    io::{self, Read, Write},
+    path::{Path, PathBuf},
+    sync::atomic::AtomicBool,
+};
+
+use itertools::Itertools;
+use tokio::{
+    fs::File,
+    io::{AsyncReadExt, AsyncWriteExt},
+};
+
+use super::{
+    segment::WALSegment,
+    util::{get_checksum, get_file_paths_with_ext, is_exist, parse_u64, validate_data, LockedFile},
+};
+
+/// Utilize thread-local variables because the tests are running concurrently,
+/// and different tests might set this value, leading to a race condition.
+#[cfg(test)]
+thread_local! {
+    static ABORT_REMOVE: AtomicBool = AtomicBool::new(false);
```

Comment:
Is this variable used to make sure there is only one task removing the file?

If yes, the access ordering should be more strict than "Relax".

Reply:
Relaxed ordering is sufficient because it is a thread local variable and requires no synchronization.

Reply:
Why do we need a thread-local atomic variable?

Reply:
The `ABORT_REMOVE` variable is used in tests to control whether the removal needs to be aborted. And the tests may runs concurrently in different threads and may interfear with each other if the variable is a single instance.